### PR TITLE
Ik 35

### PR DESCRIPTION
**Pull request by handriamanantena**

### CODE REVIEW

File: client/app/src/main/java/com/example/mrides/Notification/MatchingMessagingService.java [hunk 1]
```diff
@@ -60,21 +66,31 @@ public void onMessageReceived(RemoteMessage remoteMessage) {
...
         notificationManager.notify(0,notificationBuilder.build());
+        Notify(remoteMessage.getData());
     }
 
 
+
     @Override
     public void attach(ActivityObserver observerToAdd) {
 
+        if(observers.contains(observerToAdd)){
```
Comment (SaifMahabub) [inline client/app/src/main/java/com/example/mrides/Notification/MatchingMessagingService.java]: There should be a space between the closing parenthesis and opening curly bracket.

File: client/app/src/main/java/com/example/mrides/Notification/MatchingMessagingService.java [hunk 2]
```diff
@@ -60,21 +66,31 @@ public void onMessageReceived(RemoteMessage remoteMessage) {
...
     public void detach(ActivityObserver observerToRemove) {
-
+        observers.remove(observerToRemove);
     }
 
     @Override
-    public void Notify(String response) {
+    public void Notify() {
+
+        for(ActivityObserver e : observers){
```
Comment (SaifMahabub) [inline client/app/src/main/java/com/example/mrides/Notification/MatchingMessagingService.java]: There should be a space between the closing parenthesis and opening curly bracket.

File: client/app/src/main/java/com/example/mrides/Notification/InboxAdapter.java [hunk 3]
```diff
@@ -6,10 +6,18 @@
...
+import com.example.mrides.Activity.InboxActivity;
 import com.example.mrides.R;
 
+import java.util.Map;
+
 public class InboxAdapter extends RecyclerView.Adapter<InboxAdapter.ViewHolder>{
 
+    private Map<String,String> responseBody;
+
+    public InboxActivity(Map<String,String> responseBody){
```
Comment (SaifMahabub) [inline client/app/src/main/java/com/example/mrides/Notification/InboxAdapter.java]: There should be a space between the closing parenthesis and opening curly bracket.

File: client/app/src/main/java/com/example/mrides/Notification/InboxAdapter.java [hunk 4]
```diff
@@ -14,9 +15,11 @@
 public class InboxAdapter extends RecyclerView.Adapter<InboxAdapter.ViewHolder>{
 
     private Map<String,String> responseBody;
+    private static String inviteMessage;
 
-    public InboxActivity(Map<String,String> responseBody){
+    public InboxAdapter(Map<String,String> responseBody, String inviteMessage){
```
Comment (SaifMahabub) [inline client/app/src/main/java/com/example/mrides/Notification/InboxAdapter.java]: There should be a space between the closing parenthesis and opening curly bracket.

File: client/app/src/main/java/com/example/mrides/Notification/InboxAdapter.java [hunk 5]
```diff
@@ -15,11 +16,14 @@
 public class InboxAdapter extends RecyclerView.Adapter<InboxAdapter.ViewHolder>{
 
     private Map<String,String> responseBody;
-    private static String inviteMessage;
+    private Context inboxContext;
 
-    public InboxAdapter(Map<String,String> responseBody, String inviteMessage){
+    public InboxAdapter(Context inboxContext){
```
Comment (SaifMahabub) [inline client/app/src/main/java/com/example/mrides/Notification/InboxAdapter.java]: There should be a space between the closing parenthesis and opening curly bracket.

File: client/app/src/main/java/com/example/mrides/Notification/InboxAdapter.java [hunk 6]
```diff
@@ -15,11 +16,14 @@
...
+        this.inboxContext = inboxContext;
+    }
+
+    public void setViewComponents(Map<String,String> body){
```
Comment (SaifMahabub) [inline client/app/src/main/java/com/example/mrides/Notification/InboxAdapter.java]: There should be a space between the closing parenthesis and opening curly bracket.

File: client/app/src/main/java/com/example/mrides/Notification/InboxAdapter.java [hunk 7]
```diff
@@ -23,28 +24,25 @@ public InboxAdapter(Context inboxContext){
     }
 
     public void setViewComponents(Map<String,String> body){
```
Comment (RhythmSaif) [inline client/app/src/main/java/com/example/mrides/Notification/InboxAdapter.java]: There should be a space between the closing parenthesis and opening curly bracket.

Comment (SaifMahabub) [inline client/app/src/main/java/com/example/mrides/Notification/InboxAdapter.java]: There should be a space between the closing parenthesis and opening curly bracket.

File: client/app/src/main/java/com/example/mrides/controller/RequestHandler.java [hunk 8]
```diff
@@ -170,7 +170,7 @@ public void Notify(String response) {
 
         for(ActivityObserver e : observers){
```
Comment (SaifMahabub) [inline client/app/src/main/java/com/example/mrides/controller/RequestHandler.java]: There should be a space between the closing parenthesis and opening curly bracket.

**Pull request merged**